MekHQ/src/mekhq/gui/view/PersonViewPanel.java

```diff
@@ -740,22 +760,52 @@ private JPanel fillFamily() {
             lblSpouse2.addMouseListener(new MouseAdapter() {
                 @Override
                 public void mouseClicked(MouseEvent e) {
-                    gui.getPersonnelTab().focusOnPerson(person.getSpouseID());
+                    gui.getPersonnelTab().focusOnPerson(person.getSpouseId());
                 }
             });
-            gridBagConstraints = new GridBagConstraints();
             gridBagConstraints.gridx = 1;
             gridBagConstraints.gridy = firsty;
             gridBagConstraints.weightx = 1.0;
             gridBagConstraints.insets = new Insets(0, 10, 0, 0);
-            gridBagConstraints.fill = GridBagConstraints.NONE;
-            gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
             pnlFamily.add(lblSpouse2, gridBagConstraints);
             firsty++;
         }
 
+        if (person.hasFormerSpouse()) {
+            lblFormerSpouses1.setName("lblFormerSpouses1"); // NOI18N //$NON-NLS-1$
+            lblFormerSpouses1.setText(resourceMap.getString("lblFormerSpouses1.text")); //$NON-NLS-1$
+            gridBagConstraints = new GridBagConstraints();
+            gridBagConstraints.gridx = 0;
+            gridBagConstraints.gridy = firsty;
+            gridBagConstraints.fill = GridBagConstraints.NONE;
+            gridBagConstraints.anchor = GridBagConstraints.NORTHWEST;
+            pnlFamily.add(lblFormerSpouses1, gridBagConstraints);
+
+            gridBagConstraints.gridx = 1;
+            gridBagConstraints.weightx = 1.0;
+            gridBagConstraints.insets = new Insets(0, 10, 0, 0);
+
+            for (FormerSpouse formerSpouse : person.getFormerSpouses()) {
+                Person ex = campaign.getPerson(formerSpouse.getFormerSpouseId());
+                gridBagConstraints.gridy = firsty;
+                lblFormerSpouses2 = new JLabel();
```

Comment:
Could this be loop local instead of an instance field?

Reply:
It could be, I just find it cleaner listed above.

Reply:
I'm going to swap this out in Wave 2, and try to clean up the code here then.